src/main/java/org/dita/dost/writer/AbstractChunkTopicParser.java

```diff
@@ -300,7 +301,13 @@ Attributes processAttributes(final Attributes atts) {
                 // update @href to point back to current file
                 // if the location is moved to chunk, @href will
                 // be update again to the new location.
-                XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative + href);
+            	if(relative.toString().contains(SHARP)) {
+            		//If the relative reference already has an anchor, we need to replace that with the current anchor.
+            		final URI res = URLUtils.setFragment(relative, href.substring(1)); 
+            		XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, res.toString());
+            	} else {
+            		XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative + href);
+            	}
```

Comment:
```suggestion
            	final URI res = URLUtils.setFragment(relative, href.substring(1)); 
            	XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, res.toString());
```

Couldn't we just use `URLUtils.setFragment(relative, href.substring(1))` for both cases?

In this branch from row 299, `href` is always `#something`. The results of using `setFragment`

* `foo.dita` ->  `foo.dita#something`
* `foo.dita#old` -> `foo.dita#sommething`

Reply:
@jelovirt then how about further simplifying this entire thing:

            if (href.startsWith(SHARP)) {
            	// if @href refers to a location inside current parsing file
            	// update @href to point back to current file
            	// if the location is moved to chunk, @href will
            	// be update again to the new location.
            	final URI res = URLUtils.setFragment(relative, href.substring(1)); 
            	XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, res.toString());
            } else if (relative.toString().contains(SLASH)) {
                // if new file is not under the same directory with current file
                // add path information to the @href value
                XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative.resolve(href).toString());
            }

to this:

            if (href.startsWith(SHARP) || relative.toString().contains(SLASH)) {
                XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative.resolve(href).toString());
            }

?

Reply:
Then it would not have `setFragment` anymore. Maybe the whole code should be something like

```java
final URI hrefUri = URI.create(href);
final URI res = URLUtils.setFragment(relative.resolve(hrefURI), hrefUri.getFragment()); 
XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, res.toString());
```

Reply:
Pushed this change after adding a test case that covers both with and without fragment branches.